msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTests.java

```diff
@@ -349,6 +361,39 @@ void managedIdentity_RequestFailed_UnreachableNetwork(ManagedIdentitySourceType
         fail("MsalManagedIdentityException is expected but not thrown.");
     }
 
+    @Test
+    void azureArcManagedIdentity_MissingAuthHeader() throws Exception {
+        IEnvironmentVariables environmentVariables = new EnvironmentVariablesHelper(ManagedIdentitySourceType.AzureArc, azureArcEndpoint);
+        DefaultHttpClient httpClientMock = mock(DefaultHttpClient.class);
+
+        HttpResponse response = new HttpResponse();
+        response.statusCode(HttpStatus.SC_UNAUTHORIZED);
+
+        lenient().when(httpClientMock.send(any())).thenReturn(response);
```

Comment:
Why this lenient mocking? I added the same comment on @Avery-Dunn 's PR about MSI source. Can we get rid of it by resetting the cache?